### ISSUE TITLE
 Doctrine\DBAL\Exception\NotNullConstraintViolationException when assigning a filter to a category

### ISSUE THREAD

**Issue by medarob** [teaminmedias-pluswerk/ke_search#410]
In the search tab of a news system category I wanted to add a filter to this category. After selecting the filter and saving the form the following error appeared:

![grafik](https://user-images.githubusercontent.com/33951787/112905342-8733d480-90ea-11eb-9898-ebcf04a1d7fa.png)

The save process seems to have worked though. The filter is correctly assigned to the category inside the search tab and another filter option was automatically created. The filter option was assigned to the filter.

But the exception in the BE shouldn't happen but not sure what had caused it...

**Comment by christianbltr** [comment]
Thanks for reporting. Unfortunately I cannot reproduce this issue.

Did this appear only once or can you constantly reproduce it?

I'm not sure what could cause this issue. The value for "pid" is null, but this value is taken from the filter record which is assigned to the system category. Since you selected a filter, there should also be a pid value.

**Comment by christianbltr** [comment]
I tested this again with ke_search 4.1.0 and TYPO3 11 and could not reproduce the issue, therefore I'm closing this. Feel free to open another issue at https://github.com/tpwd/ke_search/issues if it still occurs.
